### DISCUSSION TITLE
Customise Figure separator in outline

### DISCUSSION THREAD

**Discussion by CDucloux** [typst/typst#4515] [Q&A]
In order to customise the figure caption separator in a document, I use this simple `set` rule:

```typst
#set figure.caption(
  separator: [ --- ]
)
```

Within the caption of a figure, it works as expected.

<img width="234" alt="custom_caption" src="https://github.com/typst/typst/assets/122461892/e6acf2c1-d37e-4259-8268-b251dcf003da">

However, inside the outline, the default separator is used.

<img width="391" alt="custom_outline" src="https://github.com/typst/typst/assets/122461892/697a2b61-b4d3-40f8-91f3-2278932d7171">

I couldnt find any way to customise the outline and change the separator, does the feature even exist?

Thanks!

**Comment by l0wx** [comment]
```typst
#set figure.caption(
    separator: " \u{2014} ",
)
#show outline.entry: it=> if <no-prefix> != it.fields().at(
    default: {},
    "label",
) [#outline.entry(
    it.level,
    it.element,
    {
        it.element.caption.supplement
        " "
        numbering(
            it.element.numbering,
            ..it.element.counter.at(it.element.location()),
        )
        figure.caption.separator
        it.element.caption.body
    },
    it.fill,
    it.page
)<no-prefix>] else {it}

#outline(target: figure)

#figure(
    caption: "De Dabön",
    supplement: [Dabön],
    table[Se'ăs!],
)

#figure(
    caption: "Da Gwökhod",
    ```clisp
    (print "Se'ăs!")
    ```,
)

```

**Comment by cAttte** [comment]
here's a more concise (and up-to-date) solution:
```typ
#set figure.caption(separator: " " + sym.dash.en + " ")
#show outline.entry: it => {
  it.indented(none, it.prefix() + context figure.caption.separator + it.inner())
}
```
